source/PlayerInfo.h

```diff
@@ -232,6 +237,13 @@ class PlayerInfo {
 	const ConditionsStore &Conditions() const;
 	std::map<std::string, std::string> GetSubstitutions() const;
 
+	// Get and set the "tribute" that the player receives from dominated planets.
+	bool SetTribute(const Planet * planet, int64_t payment);
+	bool SetTribute(const std::string &planetTrueName, int64_t payment);
```

Comment:
I feel like this one could be private? same for alot of these methods honestly

Reply:
Maybe, I feel that other parts of the code might just want to call those functions directly, instead of accessing them through the conditions.

Reply:
I had the feeling we generally dont do things just for planning ahead

having both gettribute and gettributeForPlanet public seems redundant?
also shouldnt it pass a reference in the method just below this line, assuming we keep it?

Reply:
Agreed, I removed "getTributeForPlanet" and changed the automatic derived conditions code to actually implement the lookup that it needs by itself.

Reply:
I also updated the code to query tribute directly, and not through conditions.

Reply:
Sounds good!